src/pages/Actions/Actions.js

```diff
@@ -0,0 +1,126 @@
+import React, { Component } from 'react';
+import { clone, setWith } from 'lodash';
+import Action from '../../components/Action/Action';
+import testHome from '../../data/testHome';
+import styled from 'styled-components';
+//styles
+const Container = styled.div`
+  width:
+  border: 1px  #ffffff solid;
+  display: grid;
+  grid-gap: 25px;
+  grid-template-columns: repeat(auto-fill, 300px);
+  
+  align-content: stretch;
+  grid-auto-flow: dense;
+
+`;
+const Item = styled.div`
+  justify-self: stretch;
+`;
+
+class Actions extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  componentDidMount() {
```

Comment:
now all the logic to open a websocket and handle messages is added to both Actions, Devices, Home and Rooms.

This should be extracted in a separate datalayer to which all these components can subscribe. But let's discuss in more detail how to do this, probably with the help of the [redux library](https://redux.js.org/)